knowledge-base/grid-inline-edit-form.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Add an Inline Form to a Grid Row
+description: How to add an inline Telerik UI for Blazor Form to a Blazor Grid row.
+type: how-to
+page_title: Add inline Telerik Form to the the Grid Rows
+slug: grid-kb-inline-form
+position: 
+tags: telerik,blazor,grid,inline,form,rows
+res_type: kb
+ticketid: 1639016
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td>Product</td>
+            <td>Grid for Blazor</td>
+        </tr>
+    </tbody>
+</table>
+
+
+## Description
+
+My end users need to be able to submit an inline Form from within a Grid row.
+
+This KB answers the following questions:
+
+* How do I show an inline Form when the user enters edit mode in the Grid?
+* How do I embed an editable inline Form inside a Grid row?
+* How do I create a cancelable inline Form inside a Grid row?
+
+
+## Solution
+
+Add an inline [Blazor Form]({%slug form-overview%}) to your Grid rows by following these steps:
+
+1. Define a [`DetailTemplate`]({%slug components/grid/features/hierarchy%}).
+1. Hide the hierarchy expand column with CSS.
+1. Use the [`SetStateAsync()` method]({%slug grid-state%}#setstateasync-examples) to enter and exit edit mode programmatically.
```

Comment:
Better link the relevant KB directly. You can do that with the "enter and ..." part, and link the method to its own documentation. In this case, leave "method" outside the link to better separate the two links.